Source/Spring2022_Capstone/Enemies/BaseEnemy.cpp

```diff
@@ -0,0 +1,46 @@
+// Created by Spring2022_Capstone team
+
+#include "BaseEnemy.h"
+#include "Spring2022_Capstone/HealthComponent.h"
+
+// Sets default values
+ABaseEnemy::ABaseEnemy()
+{
+ 	// Set this character to call Tick() every frame.  You can turn this off to improve performance if you don't need it.
+	PrimaryActorTick.bCanEverTick = true;
+
+	HealthComponent = CreateDefaultSubobject<UHealthComponent>(TEXT("HealthComponent"));
+
+	WeaponMesh = CreateDefaultSubobject<UStaticMeshComponent>(TEXT("WeaponMesh"));
+	WeaponMesh->SetupAttachment(RootComponent);
+
+	ProjectileSpawnPoint = CreateDefaultSubobject<USceneComponent>(TEXT("ProjectileSpawnPoint"));
+	ProjectileSpawnPoint->SetupAttachment(WeaponMesh);
+
+}
+
+// Called when the game starts or when spawned
+void ABaseEnemy::BeginPlay()
+{
+	Super::BeginPlay();
+	
+}
+
+void ABaseEnemy::Attack()
+{
+}
+
+// Called every frame
+void ABaseEnemy::Tick(float DeltaTime)
+{
+	Super::Tick(DeltaTime);
+
+}
+
+// Called to bind functionality to input
+void ABaseEnemy::SetupPlayerInputComponent(UInputComponent* PlayerInputComponent)
```

Comment:
Do we really need this, I assume enemies would not have any input bound to them.